### PR TITLE
[AppConfig] Drop msrest

### PR DESCRIPTION
**Pull request by YalinLi0312**
AppConfig package was regenerated with latest version of `autorest`(https://github.com/Azure/azure-sdk-for-python/pull/26375), which doesn't use `msrest`. So we can drop this requirement.

### CODE REVIEW

**Review by mccoyp — COMMENTED** [review]
LGTM! I would just wait for Xiang's approval since I'm not a package owner 🙂

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**
